src/data_readers/unit_test/data_reader_HDF5_hrrl_public_api.cpp

```diff
@@ -168,8 +168,8 @@ TEST_CASE("hdf5 data reader data field fetch tests",
 
   El::Int num_samples = 1;
 
-  auto data_store = new lbann::data_store_conduit(hdf5_dr);
-  hdf5_dr->set_data_store(data_store);
+  auto data_store = std::make_unique<lbann::data_store_conduit>(hdf5_dr.get());
+  hdf5_dr->set_data_store(data_store.get());
```

Comment:
This is tricky. The data reader is actually taking ownership (not that that's documented anywhere):
```suggestion
  {
    auto data_store = std::make_unique<lbann::data_store_conduit>(hdf5_dr.get());
    hdf5_dr->set_data_store(data_store.release());
  }
```